lua/entities/ent_jack_gmod_ezsolargenerator.lua

```diff
@@ -0,0 +1,247 @@
+AddCSLuaFile()
+
+ENT.Type = "anim"
+ENT.PrintName = "EZ Solar Panel"
+ENT.Author = "Jackarunda, AdventureBoots"
+ENT.Category = "JMod - EZ Misc."
+ENT.Information = ""
+ENT.Spawnable = true
+ENT.Base = "ent_jack_gmod_ezmachine_base"
+--
+ENT.MaxDurability = 50
+ENT.JModPreferredCarryAngles = Angle(90, 0, 0)
+ENT.MaxPower = 100
+ENT.SkyModifiers = {"clouds_", "_clouds", "cloudy_", "_cloudy", "stormy_", "storm_", "_storm"}
+--
+ENT.StaticPerfSpecs={
+	MaxDurability=50
+}
+ENT.DynamicPerfSpecs={
+	ChargeSpeed=1
+}
+function ENT:SetupDataTables()
+	self:NetworkVar("Int",0,"State")
+	self:NetworkVar("Int",1,"Grade")
+	self:NetworkVar("Float",0,"Progress")
+    self:NetworkVar("Float",1,"Visibility")
+end
+
+local STATE_BROKEN, STATE_OFF,  STATE_ON = -1, 0, 1
+
+if(SERVER)then
+    function ENT:SpawnFunction(ply,tr,ClassName)
+        local ent=ents.Create(ClassName)
+        ent:SetPos(tr.HitPos + tr.HitNormal*25)
+        ent:SetAngles(Angle(90, 90, 0))
+        JMod.Owner(ent,ply)
+        ent:Spawn()
+        ent:Activate()
+        local effectdata=EffectData()
+        effectdata:SetEntity(ent)
+        util.Effect("propspawn",effectdata)
+        return ent
+    end
+
+    function ENT:CustomInit()
+        self:SetModel("models/jmodels/props/Scaffolding_smol.mdl")
+        self:PhysicsInit(SOLID_VPHYSICS)
+        self:SetMoveType(MOVETYPE_VPHYSICS)	
+        self:SetSolid(SOLID_VPHYSICS)
+        self:DrawShadow(true)
+        self:SetUseType(SIMPLE_USE)
+        local phys = self:GetPhysicsObject()
+        if phys:IsValid() then
+            phys:Wake()
+            phys:SetMass(200)
+        end
```

Comment:
ayoooo let's make all these lines from init to here
part of the machine base
like for real my dude
let's do it
that stuff is basically boilerplate

Reply:
I was planning on doing that when we did the machine base refactor, but if you want it done now, I'll do it now.